test/datatype/reduce_local.c

```diff
@@ -44,19 +45,10 @@ static op_name_t array_of_ops [] = {
     { "replace", "MPI_REPLACE", MPI_REPLACE },
     { NULL, "MPI_OP_NULL", MPI_OP_NULL }
 };
-static int do_ops[12] = { -1, };  /* index of the ops to do. Size +1 larger than the array_of_ops */
+static int global_do_ops[12] = { -1, };  /* index of the ops to do. Size +1 larger than the array_of_ops */
```

Comment:
I doubt it is necessary, the shadowing is done in purpose.

Reply:
The shadowing creates a bunch of warnings, such as:

```
1122 reduce_local.c: In function ‘build_do_ops’:
1123 reduce_local.c:84:34: warning: declaration of ‘do_ops’ shadows a global decla\
     ration [-Wshadow]
1124  build_do_ops( char* optarg, int* do_ops)
1125                              ~~~~~^~~~~~
```

Reply:
I understand what the warning is but I said I do this on purpose. By default all functions use the global variable until I need to specialize their behavior, in which case I add a param shadowing the global variable.

Reply:
I'm not a huge fan of that paradigm, as I think it's pretty easy for developers to miss and or get wrong.  I certainly wouldn't want any of the developers in my org to write code like that.  That said, if we are going to intentionally shadow variables, can we at least document that we're doing so intentionally?  I still think it is bad form, and that we should not remove the general compiler warning, so we'll have to figure out how to ignore the warning in this file), but we at least need to tell people it was intentional.